spec/unit/util/log/destinations_spec.rb

```diff
@@ -47,24 +47,41 @@
     end
 
     describe "on POSIX systems", :if => Puppet.features.posix? do
-      let (:abspath) { '/tmp/log' }
-      let (:relpath) { 'log' }
+      describe "with a normal file" do
+        let (:abspath) { '/tmp/log' }
+        let (:relpath) { 'log' }
 
-      it_behaves_like "file destination"
+        it_behaves_like "file destination"
+
+        it "logs an error if it can't chown the file owner & group" do
+          FileUtils.expects(:chown).with(Puppet[:user], Puppet[:group], abspath).raises(Errno::EPERM)
+          Puppet.features.expects(:root?).returns(true)
+          Puppet.expects(:err).with("Unable to set ownership to #{Puppet[:user]}:#{Puppet[:group]} for log file: #{abspath}")
 
-      it "logs an error if it can't chown the file owner & group" do
-        FileUtils.expects(:chown).with(Puppet[:user], Puppet[:group], abspath).raises(Errno::EPERM)
-        Puppet.features.expects(:root?).returns(true)
-        Puppet.expects(:err).with("Unable to set ownership to #{Puppet[:user]}:#{Puppet[:group]} for log file: #{abspath}")
+          @class.new(abspath)
+        end
 
-        @class.new(abspath)
+        it "doesn't attempt to chown when running as non-root" do
+          FileUtils.expects(:chown).with(Puppet[:user], Puppet[:group], abspath).never
+          Puppet.features.expects(:root?).returns(false)
+
+          @class.new(abspath)
+        end
       end
 
-      it "doesn't attempt to chown when running as non-root" do
-        FileUtils.expects(:chown).with(Puppet[:user], Puppet[:group], abspath).never
-        Puppet.features.expects(:root?).returns(false)
+      describe "with a JSON file" do
+        let (:abspath) { '/tmp/log.json' }
+        let (:relpath) { 'log.json' }
+
+        it_behaves_like "file destination"
 
-        @class.new(abspath)
+        it "should log messages as JSON" do
+          msg = Puppet::Util::Log.new(:level => :info, :message => "don't panic")
+          json = Puppet::Util::Json.dump(msg.to_structured_hash)
+          dest = @class.new(abspath)
+          dest.handle(msg)
+          expect(File.read(abspath)).to match(%r{#{Regexp.escape(json)}})
```

Comment:
Thanks for the PR! One concern is the json log destinationuses `Puppet::Util::Json.dump`, and test basically duplicates this, it could result in false positives, e.g. if it returned an empty string, the test would still pass. I find it better to specify what the expected string is and delete the `json = ...` line above:

```ruby
expect(File.read(abspath)).to match(/"message":"don't panic"/)
```

Or even better, parse the file contents, so we know it's valid JSON:

```ruby
expect(JSON.parse(File.read(abspath))).to match({"message" => "don't panic"})
```

Reply:
Also please do retarget this PR at the 5.5.x branch, since we introduced the multi_json dependency in 5.5.0. You should be able to just edit the PR's base branch instead of creating a new PR.